onnxruntime/core/providers/cuda/tensor/scatter_nd_impl.cu

```diff
@@ -128,5 +128,193 @@ Status ScatterNDImpl(
   return Status::OK();
 }
 
+template <class T>
+struct FuncAdd {
+  __device__ __inline__ void operator()(T* start_addr, T value) const {
+    atomic_add(start_addr, value);
```

Comment:
Can we get rid of atomics?

Reply:
It is used in gather_elements_impl.cu. I know this is not the best implementation but it comes from onnxruntime itself. I'm thinking of a new one but I need to benchmark it first. I suggest keeping it for the time being to avoid memcpy device/host.